### DISCUSSION TITLE
Calendar not populated

### DISCUSSION THREAD

**Discussion by xlotlu** [photoprism/photoprism#4360] [Q&A]
On a clean install (docker compose, release 240531-60b3a4628) the calendar view doesn't get populated: 

> No albums found
> 
> Try again using other filters or keywords. Your library is continuously analyzed to automatically create albums of special moments, trips, and places.

I have tried both importing from disk and manual upload, the behaviour is similar. Detection of exif data works fine, as seen in docker log:

```
time="2024-06-27T13:43:59Z" level=info msg="import: moving main jpg file IMG_7996.JPG to 2024/02/20240213_192623_89FF0F04.jpg"
time="2024-06-27T13:44:00Z" level=info msg="media: 20240213_192623_89FF0F04.jpg was taken at 2024-02-13 19:26:23 +0000 UTC (meta)"
```

This has been reported in https://github.com/photoprism/photoprism/discussions/3364 with the same symptoms: searching by date works. It's obvious that at least part of the indexing works correctly because the All Years dropdown in search is populated with "2024".

There is only one particular thing about my install: I use MariaDB on the host, not under docker. I'm sure this is unrelated though, the database works fine:

```
$ mysql photoprism -e 'select photo_path,photo_name,photo_year,photo_month,photo_day from photos\G'
*************************** 1. row ***************************
 photo_path: 2024/02
 photo_name: 20240213_192623_89FF0F04
 photo_year: 2024
photo_month: 2
  photo_day: 13
```

**Comment by xlotlu** [comment]
@lastzero please note I am a developer, and I volunteer myself to debug and maybe even fix this, with a little bit of handholding.

I am not experienced in Go but that won't be a problem. I just need some pointers to get me started -- which part of the code I should start looking at. I guess I should first go down from the browser to the API and see what that query does, and then see if there's a matter with the underlying data.

Or help me with a shortcut to first see if the underlying data is fine.

**Comment by lastzero** [comment]
Did you wait until indexing has been completed? Do you have "Moments" enabled under Settings > General?

**Comment by xlotlu** [comment]
> Did you wait until indexing has been completed? Do you have "Moments" enabled under Settings > General?

Yes. Yes.

**Comment by lastzero** [comment]
Maybe indexing was not completed because your service/instance was killed, e.g. due to a memory limit? I can only guess at this point. As you can see on our public demo, the calendar works as expected (other users also report no issues that have not been resolved yet):
- https://demo.photoprism.app/library/calendar

Kindly note that in order for us to process bug reports, they must include a complete list of steps to reproduce a problem (with sample files if necessary) and use the bug report template we provide:
- https://www.photoprism.app/kb/reporting-bugs

Would it be alright if we moved this to [GitHub Discussions](https://github.com/photoprism/photoprism/discussions) until we know what is causing it?

**Reply by xlotlu** [reply]
> Maybe indexing was not completed because your service/instance was killed, e.g. due to a memory limit?

Docker logs showed nothing of the like. Both instances of the fail were from images in the same batch, not particularly large (5184x3456, ~6MB).

> I can only guess at this point. As you can see on our public demo, the calendar works as expected (other users also report no issues that have not been resolved yet):

Yes, except for the same symptoms under #3364 .

> Kindly note that in order for us to process bug reports, they must include a complete list of steps to reproduce a problem (with sample files if necessary) and use the bug report template we provide:
> 
>     * https://www.photoprism.app/kb/reporting-bugs

I personally find this a very interesting approach: dude offers to debug _and_ fix an issue, mentioning a new programming language won't be a problem, but following protocol is more important. Ok...

As far as the complete list of steps goes: "On a clean install", "tried both importing from disk and manual upload".

> Would it be alright if we moved this to [GitHub Discussions](https://github.com/photoprism/photoprism/discussions) until we know what is causing it?

Well, your product, your policy.


I'll just mention a plot twist: added another user, who uploaded photos from a different batch (and different camera), and they resulted in the expected behaviour.

Thank you, and regards, I will not debug this further.
